src/proc/index.rs

```diff
@@ -204,9 +209,7 @@ pub fn find_checked_indexes(
     let mut guarded_indices = BitSet::new();
 
     // Don't bother scanning if we never need `ReadZeroSkipWrite`.
-    if policies.index == BoundsCheckPolicy::ReadZeroSkipWrite
-        || policies.buffer == BoundsCheckPolicy::ReadZeroSkipWrite
-    {
+    if policies.contains(BoundsCheckPolicy::ReadZeroSkipWrite) {
```

Comment:
this appears to change the semantics a bit, since it includes a case where image policy is `ReadZeroSkipWrite` but index/buffer policies are something different
but I think it's ok, given how niche this scenario is

Reply:
Right, this is not quite needed. But it will be once I have texture bounds checks done.